src/autoformat.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import AutoformatEngine from './autoformatengine.js';
+import Feature from '../core/feature.js';
+import HeadingEngine from '../heading/headingengine.js';
+import ListEngine from '../list/listengine.js';
+
+/**
+ * The autoformat feature. Looks for predefined regular expressions and converts inserted text accordingly.
```

Comment:
This class enables autoformat feature for lists and headings. It would be good if this be described here, examples which typed characters enables which autoformatting would be nice too.